packages/product-json-to-csv/src/cli.js

```diff
@@ -0,0 +1,239 @@
+    alias: 'i',
+    default: 'stdin',
+    describe: 'Path from which to read product chunks.',
+  })
+  .coerce('input', arg => {
+    if (arg === 'stdin') return process.stdin
+
+    if (fs.existsSync(arg)) {
+      if (arg.match(/\.json$/i)) return fs.createReadStream(String(arg))
+
+      throw new Error('Invalid input file format. Must be JSON file')
+    }
+    throw new Error('Input file cannot be reached or does not exist')
+  })
+  .option('output', {
+    alias: 'o',
+    default: 'stdout',
+    describe: 'Path to output',
+    type: 'string',
+  })
+  .coerce('output', arg => {
+    if (arg !== 'stdout') return fs.createWriteStream(String(arg))
+
+    return process.stdout
+  })
+  .option('categoryBy', {
+    choices: ['name', 'key', 'externalId', 'namedPath'],
+    default: 'name',
+    describe:
+      'Define which identifier should be used for the categories column.',
+  })
+  .option('categoryOrderHintBy', {
+    choices: ['name', 'key', 'externalId', 'namedPath'],
+    default: 'name',
+    describe: 'Used CSV delimiter.',
+  })
+  .option('fillAllRows', {
+    describe:
+      'Define which identifier should be used for the categoryOrderHints column.',
+    type: 'boolean',
+  })
+  .option('language', {
+    alias: 'l',
+    default: 'en',
+    describe: 'Language used for localised attributes such as category names.',
+    type: 'string',
+  })
+  .option('delimiter', {
+    alias: 'd',
+    default: ',',
+    describe: 'Used CSV delimiter.',
+  })
+  .option('multiValueDelimiter', {
+    alias: 'm',
+    default: ';',
+    describe: 'Used CSV delimiter in multiValue fields.',
+  })
+  .option('productSeparator', {
+    alias: 'ps',
+    default: '\n',
+    describe: 'Marker separating product chunks',
+    type: 'string',
+  })
+  .option('encoding', {
+    alias: 'e',
+    default: 'utf8',
+    describe: 'Encoding used when saving data to output file',
+    type: 'string',
+  })
+  .option('logLevel', {
+    default: 'info',
+    describe: 'Logging level: error, warn, info or debug',
+    type: 'string',
+  })
+  .option('prettyLogs ', {
+    describe: 'Pretty print logs to the terminal',
+    type: 'boolean',
+  })
+  .option('logFile', {
+    default: 'product-json-to-csv.log',
+    describe: 'Path to file where logs should be saved',
+    type: 'string',
+  }).argv
+
+// instantiate logger
+const loggerConfig = {
+  level: args.logLevel,
+  prettyPrint: args.prettyLogs,
+}
+const logger = pino(loggerConfig)
+
+// print errors to stderr if we use stdout for data output
+// if we save data to output file errors are already logged by pino
+const logError = error => {
+  const errorFormatter = new PrettyError()
+
+  if (logger.level === 'debug')
+    process.stderr.write(`ERR: ${errorFormatter.render(error)}`)
+  else process.stderr.write(`ERR: ${error.message || error}`)
+}
+
+const errorHandler = errors => {
+  if (Array.isArray(errors)) errors.forEach(logError)
+  else logError(errors)
+
+  process.exitCode = 1
```

Comment:
We also sometimes needed this to be configurable in the past. Something like `continueOnErrors`. Have you thought about this?

Reply:
I'm not sure if we should continue on errors for exports. Depending on the error, it may be a bad idea. We currently continue on errors for only imports

Reply:
Hm yeah makes sense 👍